src/handlers/issue-deduplication.ts

```diff
@@ -110,32 +115,97 @@ async function handleSimilarIssuesComment(
     })
   );
 
-  let finalIndex = 0;
-  const commentBody = issueList
-    .filter((issue) =>
-      matchRepoOrgToSimilarIssueRepoOrg(payload.repository.owner.login, issue.node.repository.owner.login, payload.repository.name, issue.node.repository.name)
-    )
-    .map((issue, index) => {
-      const modifiedUrl = issue.node.url.replace("https://github.com", "https://www.github.com");
-      return `[^0${index + 1}^]: [${issue.node.title}](${modifiedUrl}) ${issue.similarity}%`;
-    })
-    .join("\n");
+  const relevantIssues = issueList.filter((issue) =>
+    matchRepoOrgToSimilarIssueRepoOrg(payload.repository.owner.login, issue.node.repository.owner.login, payload.repository.name, issue.node.repository.name)
+  );
 
-  if (commentBody.length === 0) {
+  if (relevantIssues.length === 0) {
     return;
   }
-  const footnoteLinks = [...Array(++finalIndex).keys()].map((i) => `[^0${i + 1}^]`).join("");
-  const body = "\n###### Similar " + footnoteLinks + "\n\n" + commentBody;
 
-  //Append the new foot note
+  const issueBody = context.payload.issue.body || "";
+  // Find existing footnotes in the body
+  const footnoteRegex = /\[\^(\d+)\^\]/g;
+  const existingFootnotes = issueBody.match(footnoteRegex) || [];
+  const highestFootnoteIndex = existingFootnotes.length > 0 ? Math.max(...existingFootnotes.map((fn) => parseInt(fn.match(/\d+/)?.[0] ?? "0"))) : 0;
+  let updatedBody = issueBody;
+  let footnotes: string[] | undefined;
+  relevantIssues.forEach((issue, index) => {
+    const footnoteIndex = highestFootnoteIndex + index + 1; // Continue numbering from the highest existing footnote number
+    const footnoteRef = `[^0${footnoteIndex}^]`;
+    const modifiedUrl = issue.node.url.replace("https://github.com", "https://www.github.com");
+    const { sentence } = issue.mostSimilarSentence;
+
+    // Insert footnote reference in the body
+    const sentencePattern = new RegExp(`${sentence.replace(/[.*+?^${}()|[\]\\]/g, "\\$&")}`, "g");
+    updatedBody = updatedBody.replace(sentencePattern, `${sentence}${footnoteRef}`);
+
+    // Initialize footnotes array if not already done
+    if (!footnotes) {
+      footnotes = [];
+    }
+
+    // Add new footnote to the array
+    footnotes.push(`${footnoteRef}: ⚠ ${issue.similarity}% possible duplicate - [${issue.node.title}](${modifiedUrl})\n\n`);
+  });
+
+  // Append new footnotes to the body, keeping the previous ones
+  updatedBody += footnotes ? footnotes.join("") : "";
+
+  // Update the issue with the modified body
   await context.octokit.issues.update({
     owner: payload.repository.owner.login,
     repo: payload.repository.name,
     issue_number: issueNumber,
-    body: modifiedBody + body,
+    body: updatedBody,
   });
 }
+
+/**
+ * Finds the edit distance between two strings using dynamic programming.
+ * @param sentenceA
+ * @param sentenceB
+ * @returns
+ */
+function findEditDistance(sentenceA: string, sentenceB: string): number {
+  const m = sentenceA.length;
```

Comment:
Always use full variable names. Never shorten or make ambiguous identifiers

Reply:
You can leave a comment with the original formula for other developers to refer to. A link is suitable.